src/Contract/Extra.purs

```diff
@@ -0,0 +1,124 @@
+module Contract.Extra
+  ( OneShotCurrency
+  , mintContract
+  , mintedValue
+  , oneshotPolicy
+  ) where
+
+import Prelude
+
+import Cardano.TextEnvelope (TextEnvelopeType(PlutusScriptV1))
+import Contract.Address (ownPaymentPubKeyHash, ownStakePubKeyHash)
+import Contract.Log (logInfo')
+import Contract.Monad
+  ( Contract
+  , liftContractAffM
+  , liftContractE
+  , liftContractM
+  , liftedE
+  , liftedM
+  )
+import Contract.ScriptLookups as Lookups
+import Contract.Scripts (MintingPolicy, applyArgs)
+import Contract.TextEnvelope (textEnvelopeBytes)
+import Contract.Transaction (awaitTxConfirmed, balanceAndSignTx, submit)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value (CurrencySymbol, TokenName, Value)
+import Contract.Value as Value
+import Data.Array (head)
+import Data.BigInt as BigInt
+import Data.Bifunctor (lmap)
+import Data.Either (Either, note)
+import Data.Foldable (foldMap)
+import Data.FoldableWithIndex (foldMapWithIndex)
+import Data.Generic.Rep (class Generic)
+import Data.Map as Map
+import Data.Show.Generic (genericShow)
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Newtype (unwrap, wrap)
+import Data.Tuple.Nested (type (/\), (/\))
+import Plutus.Types.Address (pubKeyHashAddress)
+import ToData (toData)
+import Types.Transaction (TransactionInput)
+import Types.PubKeyHash (PaymentPubKeyHash, StakePubKeyHash)
+
+import Types.ByteArray (ByteArray, hexToByteArray)
+import Types.Cbor (toByteArray)
+
+-- Minting policy parameterised on a txOutRef, requires the spending of said ref to passs. Makes no assertions about the tokens minted.
+-- Script size 320
+oneshotPolicyStr :: String
```

Comment:
I'd prefer not to embed this as a string, we have a new method for loading scripts, `Contract.TextEnvelope`